bokehjs/src/lib/models/widgets/file_input.ts

```diff
@@ -16,43 +19,68 @@ export class FileInputView extends WidgetView {
     if (this.dialogEl == null) {
       this.dialogEl = document.createElement('input')
       this.dialogEl.type = "file"
-      this.dialogEl.multiple = false
-      this.dialogEl.onchange = (e) => this.load_file(e)
+      this.dialogEl.multiple = false  
+      this.dialogEl.onchange = async (e) => await this.load_files(e)
       this.el.appendChild(this.dialogEl)
     }
     if (this.model.accept != null && this.model.accept != '')
       this.dialogEl.accept = this.model.accept
+    if (this.model.multiple == 'multiple' || this.model.multiple == 'True'){
+      this.dialogEl.multiple = true
+    }
     this.dialogEl.style.width = `{this.model.width}px`
     this.dialogEl.disabled = this.model.disabled
   }
 
-  load_file(e: any): void {
-    const reader = new FileReader()
-    this.model.filename = e.target.files[0].name
-    reader.onload = (e) => this.file(e)
-    reader.readAsDataURL(e.target.files[0])
-  }
-
-  file(e: any): void {
-    const file = e.target.result
-    const file_arr = file.split(",")
-
-    const content = file_arr[1]
-    const header = file_arr[0].split(":")[1].split(";")[0]
-
-    this.model.value = content
-    this.model.mime_type = header
 
+  // multiple = True
+  async load_files(e: any): Promise<void> {
+    const files = e.target.files
+    var i : number
+    var value : string[] = []
+    var filename : string[] = []
+    var mime_type : string[] = []
+    
+    for (i=0; i< files.length ; i++){
+      filename.push(files[i].name)
+      let content = await this.readfile(files[i])
+      const file_arr = content.split(",")
+      value.push(file_arr[1])
+      mime_type.push(file_arr[0].split(":")[1].split(";")[0])
+      //console.log(content);
+    }
+    if (this.model.multiple == 'multiple' || this.model.multiple == 'True'){
+      this.model.filename = filename
+      this.model.mime_type= mime_type
+      this.model.value = value
+    }
+    else {
+      this.model.filename = filename[0]
+      this.model.mime_type= mime_type[0]
+      this.model.value = value[0]
+    }
+    //console.log('load_files:')
+    //console.log(this.model.filename)
+    //console.log(this.model.value)
+  }
+  
+  readfile(file: any): Promise<string> { //<string | ArrayBuffer | null>
+    return new Promise<any>((resolve) => {
```

Comment:
All those `any` types should be refined and replaced with more specific types. If unsure about standard library types, then you get refer to [lib.dom.d.ts](https://github.com/microsoft/TypeScript/blob/01f81dfc0027e8077dfcd0fee53b94479d522a44/lib/lib.dom.d.ts#L5653) and related files.

Reply:
Actually I disliked this specification myself, it's the one "proposed" by the compiler.
I tried to use types which I considered more specific, but the compiler was never satisfied with my choice. I did not understand the background, to be honest.

Reply:
No idea how to approach the problem.
Continue Discussion in PR #9732 ?!